pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBuilderImpl.java

```diff
@@ -87,7 +87,39 @@ public ConsumerBuilderImpl(PulsarClientImpl client, Schema<T> schema) {
 
     @Override
     public ConsumerBuilder<T> loadConf(Map<String, Object> config) {
-        this.conf = ConfigurationDataUtils.loadData(config, conf, ConsumerConfigurationData.class);
+        MessageListener<T> messageListener =
+            (MessageListener<T>) config.getOrDefault("messageListener", this.conf.getMessageListener());
+        ConsumerEventListener consumerEventListener =
+            (ConsumerEventListener) config.getOrDefault("consumerEventListener", this.conf.getConsumerEventListener());
+        RedeliveryBackoff negativeAckRedeliveryBackoff = (RedeliveryBackoff) config
+            .getOrDefault("negativeAckRedeliveryBackoff", this.conf.getNegativeAckRedeliveryBackoff());
+        RedeliveryBackoff ackTimeoutRedeliveryBackoff = (RedeliveryBackoff) config
+            .getOrDefault("ackTimeoutRedeliveryBackoff", this.conf.getAckTimeoutRedeliveryBackoff());
+        CryptoKeyReader cryptoKeyReader =
+            (CryptoKeyReader) config.getOrDefault("cryptoKeyReader", this.conf.getCryptoKeyReader());
+        MessageCrypto messageCrypto =
+            (MessageCrypto) config.getOrDefault("messageCrypto", this.conf.getMessageCrypto());
+        BatchReceivePolicy batchReceivePolicy =
+            (BatchReceivePolicy) config.getOrDefault("batchReceivePolicy", this.conf.getBatchReceivePolicy());
+        KeySharedPolicy keySharedPolicy =
+            (KeySharedPolicy) config.getOrDefault("keySharedPolicy", this.conf.getKeySharedPolicy());
+        MessagePayloadProcessor payloadProcessor =
+            (MessagePayloadProcessor) config.getOrDefault("payloadProcessor", this.conf.getPayloadProcessor());
+
+        ConsumerConfigurationData<T> configurationData =
+            ConfigurationDataUtils.loadData(config, conf, ConsumerConfigurationData.class);
+
+        configurationData.setMessageListener(messageListener);
+        configurationData.setConsumerEventListener(consumerEventListener);
+        configurationData.setNegativeAckRedeliveryBackoff(negativeAckRedeliveryBackoff);
+        configurationData.setAckTimeoutRedeliveryBackoff(ackTimeoutRedeliveryBackoff);
```

Comment:
Is it worth just making these two (and some of the others) serializable?

Reply:
The hardest part is making them work with Jackson. Not sure it's possible for `RedeliveryBackoff`

Reply:
I believe RB can be serialized by configuring a mix-in (on the mapper) that provides a json creator constructor for RB. Not saying that it is ideal, just that it can be done.

Reply:
RB being an interface. I'm not sure it can be done. How to decide which impl to use ?

Reply:
In that case we could use `@JsonTypeInfo` + `@JsonSubTypes` in a mix-in. This would work well as long as we know about all of the RB impls. 

NOTE: Whatever we do for RB could apply to the `ClientConfigurationData.authentication` as it  also currently ignored and is also an interface. There may be other fields as well.